### PR TITLE
[ci] Optimize build time and number of runners

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do ?
Optimize Travis CI configuration by:
- Stop using push build on PRs.
- Disable Arm jobs on PRs

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -42,58 +42,59 @@ jobs:
...
 # Node 6 - x64
 # ---------------------------------------
-    - stage: Functional tests
-      name: "Functionnal Tests - amd64 - Node v6"
+    - stage: "E2E Tests"
+      name: "amd64 - Node v6"
+
+      if: |
+        (type = push AND branch =~ ^(master|[0-9]+-dev)$) OR
+        (type = pull_request AND branch = master) OR
```
Comment (jenow) [inline .travis.yml]: Question: Shouldn't we still run the e2e tests with node v6 even on pull request?

File: .travis.yml [hunk 2]
```diff
@@ -1,29 +1,15 @@
...
 jobs:
   include:
 # ---------------------------------------
 # Unit Testing
 # ---------------------------------------
-    - stage: Unit tests & Linters
-      name: "Unit Tests"
-
+    - stage: "Tests & Linters"
+      if: type = pull_request OR type = cron OR (type = push AND branch =~ ^master|.*-dev$)
```
Comment (scottinet) [inline .travis.yml]: Missing stable branches for previous, still maintained major versions:

```suggestion
      if: type = pull_request OR type = cron OR (type = push AND branch =~ ^master|.*-dev|.*-stable$)
```

File: .travis.yml [hunk 3]
```diff
@@ -40,60 +27,57 @@ jobs:
 # ---------------------------------------
 # SonarQube
 # ---------------------------------------
-    - stage: Unit tests & Linters
+    - stage: "Tests & Linters"
+      if: (type = pull_request AND fork != true) OR (type = push AND branch =~ ^master|.*-dev$)
```
Comment (scottinet) [inline .travis.yml]: same here

```suggestion
      if: (type = pull_request AND fork != true) OR (type = push AND branch =~ ^master|.*-dev|.*-stable$)
```

Comment (scottinet) [reply .travis.yml]: I'll let you make the other changes

File: .travis.yml (hunk 2 above)
Comment (alexandrebouthinon) [reply .travis.yml]: Yes thank you :+1:

**Comment by alexandrebouthinon** [comment]
@scottinet: Thanks for your feedback, I also add NPM release deployments on `*-stable` branches

**Pull request merged**